.github/workflows/third-party.yml

```
@@ -113,18 +113,82 @@ jobs:
     - name: Run SQLModel tests
       run: uv run --no-project pytest tests
 
+  test-beanie:
+    name: Test Beanie (main branch) on Python ${{ matrix.python-version }}
+    # If 'schedule' was the trigger, don't run it on contributors' forks
+    if: |
+      github.event_name == 'workflow_dispatch' ||
+      (github.event_name == 'schedule' && github.repository == 'pydantic/pydantic') ||
+      (github.event_name == 'pull_request' && contains(github.event.pull_request.labels.*.name, 'third-party-tests'))
+
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [ "3.8", "3.9", "3.10", "3.11", "3.12", "3.13" ]
+        mongodb-version: [4.4, 5.0, 6.0, 7.0, 8.0 ]
+    steps:
+    - name: Checkout beanie
+      uses: actions/checkout@v4
+      with:
+        repository: BeanieODM/beanie
+
+    - name: Checkout Pydantic
+      uses: actions/checkout@v4
+      with:
+        path: pydantic-latest
+
+    - uses: astral-sh/setup-uv@v4
+      with:
+        enable-cache: true
+        python-version: ${{ matrix.python-version }}
```

```suggestion
    - uses: actions/setup-python@v5
      with:
        python-version: ${{ matrix.python-version }}
```

Best to be as close to the project's CI.